### PR TITLE
improved postprocessing scaling

### LINKED ISSUE #857: Improve scaling of postprocessing tasks
**Is your feature request related to a problem? Please describe.**
some scaling experiments on supercomputers allowed us to spot some execution bottlenecks at the postprocessing level. In this case, when the postprocessing is called at the end of a caprieval-free workflow, the parallelisation is not great, as it takes the parameter proper to the batch mode execution

**Describe the solution you'd like**
It should be easy to call the postprocessing with the current running parameters (mode + number of cores) from within haddock3

**Additional context**
This should save a lot of CPU time in these occasions. Goes also along the Bioexcel lines of improving the scaling of core applications.

### PR DESCRIPTION
**Pull request by mgiulini**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [ ] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [ ] Your code follows our coding style
- [ ] You wrote tests for the new code
- [ ] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [ ] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [ ] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Closes #857 by improving the scaling of the postprocessing analysis.

1. When caprieval folders are not present in the workflow, the postprocessing analysis uses the current mode and cores in the CAPRI calculations
2. pre-caprieval model unpacking (and post caprieval model compression) receive the same parameters

PS: the comparison between the results of `analysis/topoaa-clustfcc-test.cfg` as it is impossible to get full reproducibility in that case

### CODE REVIEW

File: src/haddock/clis/cli_analyse.py [hunk 1]
```diff
@@ -183,6 +183,26 @@ def update_paths(
...
+    choices=["local", "batch", "mpi"],
+)
+
+ap.add_argument(
+    "--ncores",
+    help="number of cores to use",
+    required=False,
+    type=int,
+    default=1,
+    # maximum allowed value for ncores is 16
```
Comment (VGPReys) [inline src/haddock/clis/cli_analyse.py:201]: why is it limited to 16 ?

**Review by VGPReys — APPROVED** [review]
Ready to be merged ! 🚀

File: src/haddock/clis/cli_analyse.py (hunk 1 above)
Comment (mgiulini) [reply src/haddock/clis/cli_analyse.py:201]: good question..cannot remember a good reason for that, I will put something like 100

**Comment by amjjbonvin** [comment]
Isn’t that defined in a default.yaml file?

On the lumi supercomputer I changed that value to the max number of cores per node

**Comment by mgiulini** [comment]
> Changing it in the yaml file did speed up the analysis when tested in lumi - so it seems there is a max defined/used

it used to depend on the run before: when caprieval was run within the workflow the scaling was noticeable, otherwise, if no caprieval was run, the postprocessing was launching capri calculations using few cores..with this PR there will not be any difference anymore

**Pull request merged**